commands/publish/index.js

```diff
@@ -578,12 +578,29 @@ class PublishCommand extends Command {
     }
   }
 
+  runPrepublishScripts(pkg) {
+    return Promise.resolve(this.runPackageLifecycle(pkg, "prepublish"))
+      .then(() => this.runPackageLifecycle(pkg, "prepare"))
+      .then(() => this.runPackageLifecycle(pkg, "prepublishOnly"));
+  }
+
   npmPublish() {
     const tracker = this.logger.newItem("npmPublish");
     // if we skip temp tags we should tag with the proper value immediately
     const distTag = this.options.tempTag ? "lerna-temp" : this.getDistTag();
 
-    this.updates.forEach(({ pkg }) => this.execScript(pkg, "prepublish"));
+    const rootPkg = this.project.package;
+
+    const promise = Promise.resolve()
+      .then(() => this.runPrepublishScripts(rootPkg))
+      .then(() =>
+        Promise.all(
+          this.updates.map(({ pkg }) => {
```

Comment:
Instead of `Promise.all()`, this should be `pMap(this.updates, ({ pkg }) => {...})`. `this.updates` will likely be a `Set` soon, and we may need to tune the concurrency.